modules/loki-rate-limit-errors.adoc

```diff
@@ -0,0 +1,71 @@
+// Module is included in the following assemblies:
+// * logging/cluster-logging-loki.adoc
+// * logging/cluster-logging-external.adoc
+
+:_content-type: PROCEDURE
+[id="loki-rate-limit-errors_{context}"]
+= Troubleshooting Loki rate limit errors
+
+If the Log Forwarder API forwards a large block of messages to Loki that exceeds the rate limit, Loki generates rate limit (`429`) errors. You can fix this issue by modifying the `LokiStack` custom resource (CR).
+
+[IMPORTANT]
+====
+The `LokiStack` CR is not available on Grafana-hosted Loki. This topic does not apply to Grafana-hosted Loki servers.
+====
+
+.Conditions
+
+* The Log Forwarder API is configured to forward logs to Loki.
+
+* Your system sends a block of messages that is larger than 2 MB to Loki. For example:
++
+[source,text]
+----
+"values":[["1630410392689800468","{\"kind\":\"Event\",\"apiVersion\":\
+.......
+......
+......
+......
+\"received_at\":\"2021-08-31T11:46:32.800278+00:00\",\"version\":\"1.7.4 1.6.0\"}},\"@timestamp\":\"2021-08-31T11:46:32.799692+00:00\",\"viaq_index_name\":\"audit-write\",\"viaq_msg_id\":\"MzFjYjJkZjItNjY0MC00YWU4LWIwMTEtNGNmM2E5ZmViMGU4\",\"log_type\":\"audit\"}"]]}]}
+----
+
+* After you enter `oc logs -c collector`, the collector logs in your {ocp-logging} cluster show the following error messages:
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded (limit: 8388608 bytes/sec) while attempting to ingest '2140' lines totaling '3285284' bytes
+----
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded' or '500 Internal Server Error rpc error: code = ResourceExhausted desc = grpc: received message larger than max (5277702 vs. 4194304)'
```

Comment:
Do we need the above 429 messages when there is one below too?

Reply:
Maybe not? I just didn't remove them because they were pre-existing in the docs. I am happy to take them out if it makes sense to.

Reply:
I agree that this looks a bit duplicated, but I think I understand the purpose: first explaining what to look for in the output and then providing concrete examples. I have added a suggestion (covering a whole lot of lines) in another comment.